test/EndToEnd/HydraUtils.hs

```diff
@@ -28,61 +29,74 @@ import CardanoNode (
+import Hydra.Cluster.Faucet (Marked (..), publishHydraScriptsAs, seedFromFaucet_)
 import Hydra.Cluster.Fixture qualified as HydraFixture
-import HydraNode (HydraClient (..))
+import Hydra.ContestationPeriod (
+  ContestationPeriod (UnsafeContestationPeriod),
+ )
+import HydraNode (EndToEndLog (..), HydraClient (..), waitForNodesConnected, withHydraCluster)
+import Test.Hydra.Prelude (withTempDir)
 
 -- HydraAuction imports
 
-import Hydra.Cardano.Api (UTxO' (UTxO))
+import Hydra.Cardano.Api (TxId, UTxO' (UTxO))
+import HydraAuction.Runner (ExecutionContext (MkExecutionContext, node, tracer), HydraAuctionLog (FromHydra), Runner, executeRunner)
 import HydraAuctionUtils.Fixture (
   Actor (..),
+  hydraKeysFor,
   hydraNodeActors,
+  keysFor,
  )
 
-prepareScriptRegistry :: RunningNode -> IO ScriptRegistry
+prepareScriptRegistry :: RunningNode -> IO (TxId, ScriptRegistry)
 prepareScriptRegistry node@RunningNode {networkId, nodeSocket} = do
   hydraScriptsTxId <-
     liftIO $ publishHydraScriptsAs node HydraFixture.Faucet
-  queryScriptRegistry networkId nodeSocket hydraScriptsTxId
+  scriptRegistry <- queryScriptRegistry networkId nodeSocket hydraScriptsTxId
+  pure (hydraScriptsTxId, scriptRegistry)
+
+spinUpHeads :: Int -> TxId -> (ThreeClients -> Runner ()) -> Runner ()
+spinUpHeads clusterIx hydraScriptsTxId cont = do
+  liftIO $ setEnv "HYDRA_CONFIG_DIR" "./data"
+  ctx@(MkExecutionContext {node, tracer}) <- ask
+  let hydraTracer = contramap FromHydra tracer
+  liftIO $
+    withTempDir "end-to-end-test" $ \tmpDir -> do
+      oscarKeys@(oscarCardanoVk, _) <- keysFor Oscar
+      patriciaKeys@(patriciaCardanoVk, _) <- keysFor Patricia
+      rupertKeys@(rupertCardanoVk, _) <- keysFor Rupert
+
+      (_, oscarSk) <- hydraKeysFor Oscar
+      (_, patriciaSk) <- hydraKeysFor Patricia
+      (_, rupertSk) <- hydraKeysFor Rupert
+
+      let cardanoKeys = [oscarKeys, patriciaKeys, rupertKeys]
+          hydraKeys = [oscarSk, patriciaSk, rupertSk]
+
+      let firstNodeId = clusterIx * 3
+      let contestationPeriod = UnsafeContestationPeriod 2
+      withHydraCluster
+        hydraTracer
+        tmpDir
+        (nodeSocket node)
+        firstNodeId
+        cardanoKeys
+        hydraKeys
+        hydraScriptsTxId
+        contestationPeriod
+        $ \nodes -> do
+          [n1, n2, n3] <- pure $ toList nodes
+          waitForNodesConnected hydraTracer [n1, n2, n3]
+          -- Funds to be used as fuel by Hydra protocol transactions
+          let faucetTracer = contramap FromFaucet hydraTracer
+          seedFromFaucet_ node oscarCardanoVk 100_000_000 Fuel faucetTracer
+          seedFromFaucet_ node patriciaCardanoVk 100_000_000 Fuel faucetTracer
+          seedFromFaucet_ node rupertCardanoVk 100_000_000 Fuel faucetTracer
+          executeRunner ctx $ cont ((n1, Oscar), (n2, Patricia), (n3, Rupert))
 
 type ThreeClients =
   ((HydraClient, Actor), (HydraClient, Actor), (HydraClient, Actor))
 
-runningThreeNodesDockerComposeHydra ::
```

Comment:
Please do not remove this, I wanna to have ability to check tests on docker-compose locally.

Reply:
@uhbif19 I'm afraid this PR already breaks current docker setup, as it moves around some files in the `data` folder (this is needed, as the hydra nodes assumes a certain structure, you can only tell it the root folder in which to look for config files). We could definitely get the docker setup working again by updating the compose file and the scripts to respect the new structure, but it requires some extra effort.

Reply:
I created #273 to track this issue